app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -17,15 +17,16 @@ def authenticate
         logger.info("Authorized user #{user.login}(#{user.to_label})")
         set_current_user user
       else
-        if api_request?
-          false
-        else
-          # Keep the old request uri that we can redirect later on
-          session[:original_uri] = request.fullpath
-          @available_sso ||= SSO::Base.new(self)
-
-          redirect_to @available_sso.login_url unless @available_sso.has_rendered
+        return false if api_request?
+        # Keep the old request uri that we can redirect later on
+        session[:original_uri] = request.fullpath
+        @available_sso ||= SSO::Base.new(self)
+        return if @available_sso.has_rendered
```

Comment:
I created a test that demonstrates the issue if you'd like to include it: https://github.com/domcleal/foreman/commit/9cf396bb7867570485bd63e6f2b6cb5cd21a5629